pkg/products/monitoringspec/reconciler_test.go

```diff
@@ -0,0 +1,514 @@
+package monitoringspec
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/integr8ly/integreatly-operator/pkg/config"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+
+	prometheusmonitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+
+	applicationmonitoringv1alpha1 "github.com/integr8ly/application-monitoring-operator/pkg/apis/applicationmonitoring/v1alpha1"
```

Comment:
I think what I was saying about this not being needed might still be valid: yes it's referenced here in a couple of places, but I think it can be removed from those places:

* I _think_ it shouldn't be needed in the scheme, since we're not going to be making any requests related to types from that apigroup in this test
* Where it's referenced in the `TypeMeta` of `InstallPlanList` instances, I think that's actually incorrect. I'd suggest just completely removing the `TypeMeta` field from those instances, as it should get set correctly implicitly.

Reply:
The basic things were first copied from monitoring-middleware unit tests. I removed the import and corrected the scheme.
```TypeMeta``` in the install plan list has been removed as suggested.

Reply:
Also removed, the unnecessary scheme additions.